R/regression.R

```diff
@@ -422,15 +422,25 @@ Regression <- function(formula = as.formula(NULL),
             labels.to.update <- FALSE
         else
             labels.to.update <- vars.with.data.refs
-        relabelled.outputs <- relabelFormulaAndData(new.var.names, input.formula, data, update.labels = labels.to.update)
+        if (any(labels.to.update))
+        {
+            new.labels <- paste0(original.labels[labels.to.update], " from ", dataset.names[labels.to.update])
+            data <- updateAttribute(data, attr.to.update = "label", updated.values = new.labels)
+        }
+        syntactic.new.names <- make.names(new.var.names, unique = TRUE)
+        if (!identical(unname(new.var.names), syntactic.new.names))
+            names(syntactic.new.names) <- all.variable.names
+        else
+            syntactic.new.names <- new.var.names
+        relabelled.outputs <- relabelFormulaAndData(syntactic.new.names, input.formula, data)
         input.formula <- relabelled.outputs$formula
         data <- relabelled.outputs$data
         all.variable.names <- AllVariablesNames(input.formula, data = data)
     }
     non.syntactic.names <- checkForNonSyntacticNames(all.variable.names)
     if (non.syntactic.names.exist <- !is.null(non.syntactic.names))
     {
```

Comment:
Can this block now be moved to an `else if`  of the `if` above?

Reply:
The code for that would look like this
` else if (non.syntactic.names.exist <- !is.null(non.syntactic.names <- checkForNonSyntacticNames(all.variable.names)))` which is quite long and assigns two variables in one line too.

Reply:
`non.syntactic.names.exist` is used as a flag to check if the interaction formula needs to be updated too

Reply:
The code shouldnt need to enter the second if and call `relabelFormulaAndData` if it already entered the first `if` since the names are forced to be syntactic inside the first `if` statement